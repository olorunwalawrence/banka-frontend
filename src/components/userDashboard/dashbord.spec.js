import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { mount, toJson } from '../../../test/enzyme';
import Dashboard, { fillTable } from './dashboard';

let user = {
  id: 2,
  firstname: 'Prince',
  lastname: 'mike',
  email: 'mike@email.com',
  type: 'user',
  isadmin: false,
}
const accounts = [{
  id: 1,
  accountnumber: 553373313,
  createdon: "2019-08-12T00:00:00.000Z",
  owner: 2,
  owneremail: "mike@email.com",
  type: "savings",
  status: "active",
  balance: 0,
}]
const context = {
  auth: {
    isLoading: false,
    user,
    error: ''
  },
  account: {
    success: '',
    error: '',
    accountDetails: {},
    accounts,
  }
}
const store = {
  getState: () => ({ ... context }),
  subscribe: () => ({}),
  dispatch: () => ({
    type: 'GET_ACCOUNT', 
    payload: {
      success: '',
      error: '',
      accountDetails: {},
      accounts,
      noAccount: '',
    }
  })
}

describe('Layout Test', () => {
  let wrapper;
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
      <Router>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </Router>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Dashboard />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('UserHeader')).toBeTruthy();
    expect(wrapper.find('.drop-menu')).toBeTruthy();
    expect(wrapper.find('.two-third')).toBeTruthy();
  });
  it('should display the layout for creating an account properly', () => {
    context.auth.user = {};
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Dashboard />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('Redirect').props().to).toEqual('/login');
  });
  it('should display the layout for dashboard properly', () => {
    const list = fillTable(accounts, user);
    
    expect(list[0].props.children[0].type).toEqual('ul');
    expect(list[0].props.children[0].key).toEqual(`${accounts[0].accountnumber}`);
  });
});
