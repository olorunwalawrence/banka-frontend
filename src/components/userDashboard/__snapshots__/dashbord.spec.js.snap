// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Test should display the layout for creating an account properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Dashboard)>
        <Dashboard
          accounts={
            Array [
              Object {
                "accountnumber": 553373313,
                "balance": 0,
                "createdon": "2019-08-12T00:00:00.000Z",
                "id": 1,
                "owner": 2,
                "owneremail": "mike@email.com",
                "status": "active",
                "type": "savings",
              },
            ]
          }
          checkLoggedUser={[Function]}
          getAccount={[Function]}
          isLoading={false}
          logout={[Function]}
          user={
            Object {
              "email": "mike@email.com",
              "firstname": "Prince",
              "id": 2,
              "isadmin": false,
              "lastname": "mike",
              "type": "user",
            }
          }
        >
          <Connect(UserHeader)>
            <UserHeader
              logout={[Function]}
            >
              <div
                className="container"
                id="heed"
              >
                <h1
                  className="white"
                >
                  <span>
                    <button
                      className="drop-butn nn hiden"
                      onMouseDown={[Function]}
                      type="button"
                    >
                      <i
                        className="icofont-navigation-menu icofont-2x"
                      />
                    </button>
                    MyBanka
                  </span>
                </h1>
              </div>
              <ul
                className=" drop-menu hide"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Link
                  to="/home"
                >
                  <a
                    href="/home"
                    onClick={[Function]}
                  >
                    <li>
                      Dashboard
                    </li>
                  </a>
                </Link>
                <Link
                  to="/account"
                >
                  <a
                    href="/account"
                    onClick={[Function]}
                  >
                    <li>
                      Create Account
                    </li>
                  </a>
                </Link>
                <Link
                  onClick={[Function]}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <li>
                      Logout
                    </li>
                  </a>
                </Link>
              </ul>
            </UserHeader>
          </Connect(UserHeader)>
          <main
            className="flex"
          >
            <div
              className="one-third aside"
            >
              <ul
                className="user"
              >
                <li>
                  <Link
                    to="/home"
                  >
                    <a
                      href="/home"
                      onClick={[Function]}
                    >
                      Dashboard
                    </a>
                  </Link>
                </li>
                <li>
                  <Link
                    to="/account"
                  >
                    <a
                      href="/account"
                      onClick={[Function]}
                    >
                      Create Account
                    </a>
                  </Link>
                </li>
                <li>
                  <Link
                    onClick={[Function]}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Logout
                    </a>
                  </Link>
                </li>
              </ul>
            </div>
            <div
              className="two-third main"
            >
              <div />
            </div>
          </main>
        </Dashboard>
      </Connect(Dashboard)>
    </Provider>
  </Router>
</BrowserRouter>
`;
